CMakeLists.txt

```diff
@@ -21,19 +21,54 @@ if (NOT CMAKE_BUILD_TYPE)
             "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
 endif ()
 
+# Set build options
+option (BUILD_UFS "Build GOCART component for UFS" OFF)
+
 set (DOING_GEOS5 YES)
 
 # Should find a better place for this - used in Chem component
 set (ACG_FLAGS -v)
 set (FV_PRECISION R4R8)
 
+if (BUILD_UFS)
+# Ensure we build as 32-bit
+  message ("Force 32-bit build for GOCART")
+  if (CMAKE_Fortran_COMPILER_ID MATCHES "Intel")
+    string (REPLACE "-real-size 64" "" CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
+  elseif (CMAKE_Fortran_COMPILER_ID MATCHES "GNU")
+    string (REPLACE "-fdefault-real-8" "" CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS}")
+  endif()
+endif()
+
 if (NOT COMMAND esma)
     list (APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake@")
     include (esma)
 endif()
 
     ecbuild_declare_project()
 
+# Find dependencies
+if (NOT NetCDF_Fortran_FOUND)
+  find_package (NetCDF REQUIRED Fortran)
```

Comment:
These `'find_package` calls are causing cmake errors for me when I try to build GEOS using this branch. They should be wrapped in a `if (BUILD_UFS)` or `if (NOT Baselibs_FOUND)`? @mathomp4 suggested that it maybe best to have both.

Reply:
Yeah. My thought was the `find_package` commands could be in an `if (NOT Baselibs_FOUND)` block as is done in MAPL (https://github.com/GEOS-ESM/MAPL/blob/main/CMakeLists.txt).

Meanwhile the flag options could be in a more UFS specific if-block

Reply:
I agree with @mathomp4 - we can add a `if (NOT Baselibs_FOUND)` block wrapping all Baselibs-specific `find_package` commands.

The flag option has been wrapped with `if (BUILD_UFS)` in commit d06f07d0.

Reply:
@mathomp4, @WilliamJamieson - could you please provide directions on how to build GOCART with Baselibs, as done in GEOS? I'd like to test this change before submitting it.

Reply:
@rmontuoro I think we could help. Baselibs is (usually nowadays) pretty simple to build, though every new machine seems to throw its own special wrench. But a set up of GNU and Open MPI should get you most of the way there (maybe MKL or BLAS/LAPACK is needed for GEOS? MKL is free, so I often forget about it).

Reply:
Commit a9c1314 wraps all `find_package` calls with a `if (NOT Baselibs_FOUND)` block, as agreed.

After these changes, GOCART builds successfully on Orion both within UFS-GOCART and following @WilliamJamieson's [instructions](https://github.com/GEOS-ESM/GOCART/wiki/Building-GOCART-on-Orion ), which use Baselibs. Note, however, that in the latter case the included legacy GOCART (`GOCART_GridComp`) did not build since it requires additional chemistry modules. Also, MAPL should be added as an explicit dependency in the `CMakeLists.txt` file for each of the legacy GOCART components. For example:

```
diff --git a/ESMF/GOCART_GridComp/BC_GridComp/CMakeLists.txt b/ESMF/GOCART_GridComp/BC_GridComp/CMakeLists.txt
index a26a18c..bfb9224 100644
--- a/ESMF/GOCART_GridComp/BC_GridComp/CMakeLists.txt
+++ b/ESMF/GOCART_GridComp/BC_GridComp/CMakeLists.txt
@@ -2,7 +2,7 @@ esma_set_this ()
 
 esma_add_library (${this}
   SRCS ${this}Mod.F90
-  DEPENDENCIES Chem_Shared Chem_Base GMAO_mpeu MAPL_cfio_r4)
+  DEPENDENCIES Chem_Shared Chem_Base GMAO_mpeu MAPL MAPL_cfio_r4)
 target_include_directories (${this} PUBLIC ${INC_ESMF} ${INC_NETCDF})
```

Reply:
@rmontuoro this is an excellent point. I will look into making these changes.